client/components/product-icon/index.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isBloggerPlan,
+	isPersonalPlan,
+	isPremiumPlan,
+	isBusinessPlan,
+	isEcommercePlan,
+	getPlan,
+	getPlansKeys,
+	getPlanClass,
+} from 'lib/plans';
+import {
+	getProductClass,
+	getProductsKeys,
+	getProductShortName,
+	isJetpackBackupSlug,
+	isJetpackPlanSlug,
+} from 'lib/products-values';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export default class ProductIcon extends Component {
+	getPlanIcon( iconName ) {
```

Comment:
Is there a reason why we're not using `<PlanIcon />` here?

Reply:
I'm planning on eventually replacing `PlanIcon` with `ProductIcon` globally. This component is an enhanced version of `PlanIcon`.

The reason why those enhancement did not land inside `PlanIcon` is that we want to change its name to something more generic.